### PR TITLE
#170947426  Setup dotenv environment variables

### PR DESCRIPTION
**Pull request by hakoemmy**
#### What does this PR do?
- Have dotenv setup working
#### Description of Task to be completed?
 Allow the following packages to fetch the current environment variables
 - Sequelize
 -  Passport and among others
#### How should this be manually tested?
- Just clone the [repo](https://github.com/andela/octopus-bn-backend.git) and 
- follow the instructions in [ReadMe](https://github.com/andela/octopus-bn-backend/blob/develop/README.md)
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#170947426](https://www.pivotaltracker.com/story/show/170947426)
### Screenshots (if appropriate)
N/A
#### Questions:
N/A

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,3 +5,19 @@ Barefoot Nomad - Making company travel and accomodation easy and convinient.
...
+---
+### Dotenv setup
+ 1. ***Get some stuffs on table***
+  * Run ``` npm install ```
+  * Create ``` .env ``` in project root directory
+  * Take a look on ``` env.example.js ``` which is in project root directory to have a clue on environment variables that are already being used in project
+  * Copy all keys from ``` .env.example.js ``` to ``` .env ``` and add values to corresponding keys
+  * Feel free to add other keys and values according to your feature requirements
+  ***Note***: Add keys in ``` .env.example.js ``` to ease next setup for other developers.
+
```
Comment (izzett222) [inline README.md]: hey @hakoemmy ,  is it env.example.js file or env.example file?

**Review by izzett222 — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (AbdoulNuru) [reply README.md]: Yes @izzett222  I think it is .env.example, @hakoemmy You can change the name in the readme

Comment (hakoemmy) [reply README.md]: @izzett222 @AbdoulNuru Thank you for that, allow me to fix that

Comment (hakoemmy) [reply README.md]: @AbdoulNuru @izzett222 I hope I fixed the issue like so.

**Review by AbdoulNuru — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (AbdoulNuru) [reply README.md]: @hakoemmy Now it is okay for me

**Review by blaise82 — APPROVED** [review]

**Review by RusimbiPatrick — APPROVED** [review]
I think we will need a dot

**Review by RusimbiPatrick — APPROVED** [review]

**Review by izzett222 — APPROVED** [review]

File: src/index.js [hunk 2]
```diff
@@ -1,93 +1,13 @@
...
-            error: {}
-        }
-    });
-});
-
-// finally, let's start our server...
-const server = app.listen(process.env.PORT || 3000, function() {
-    console.log("Listening on port " + server.address().port);
+app.listen(process.env.PORT, () => {
+  console.log(`Server is running on PORT ${process.env.PORT}....`);
```
Comment (kcharles52) [inline src/index.js]: The code you have remove is required in the project. Just set up only the donenv environment and the areas it affects and leave the code that is not affected

File: src/index.js [hunk 3]
```diff
@@ -1,93 +1,13 @@
...
 });
+export default app;
```
Comment (kcharles52) [inline src/index.js]: Make sure you add a blank line at the end of the file
```suggestion
export default app;

```

File: README.md [hunk 4]
```diff
@@ -5,3 +5,19 @@ Barefoot Nomad - Making company travel and accomodation easy and convinient.
 Make company global travel and accommodation easy and convenient for the strong workforce of savvy members of staff, by leveraging the modern web.
 
 ---
+## Project setup
+---
+### Dotenv setup
+ 1. ***Get some stuffs on table***
+  * Run ``` npm install ```
+  * Create ``` .env ``` in project root directory
+  * Take a look on ``` .env.example ``` which is in project root directory to have a clue on environment variables that are already being used in project
```
Comment (kcharles52) [inline README.md]:
```suggestion
  * Take a look at the ``` .env.example ```  file which is in the project root directory to have a clue on environment variables that are being used in this project
```

File: README.md [hunk 5]
```diff
@@ -5,3 +5,19 @@ Barefoot Nomad - Making company travel and accomodation easy and convinient.
...
+  * Copy all keys from ``` .env.example ``` to ``` .env ``` and add values to corresponding keys
```
Comment (kcharles52) [inline README.md]:
```suggestion
  * Copy all keys from the ``` .env.example ```  file to ``` .env ``` file and add values to corresponding keys. These can be obtained from the project administrator
```

File: README.md [hunk 6]
```diff
@@ -5,3 +5,19 @@ Barefoot Nomad - Making company travel and accomodation easy and convinient.
...
+  * Feel free to add other keys and values according to your feature requirements
+  ***Note***: Add keys in ``` .env.example ``` to ease next setup for other developers.
+
+  2. ***Time to serve***
+   * For you to use your new environment variable you have to import ``` dotenv ``` file anywhere you want to utulise your environment variables and configure it. like this: ``` import dotenv from 'dotenv';dotenv.config(); ```
```
Comment (kcharles52) [inline README.md]:
```suggestion
   * For you to use your new environment variable you have to import ``` dotenv ``` modlue in the file where you want to utilise your environment variables and configure it. like this: 
   ```import dotenv from 'dotenv';
   dotenv.config();```
```

File: .env.example [hunk 7]
```diff
@@ -0,0 +1,7 @@
+DATABASE_DEV_URL=''
+DATABASE_TEST_URL=''
+JWT_SECRET=''
+FACEBOOK_APP_ID=''
+FACEBOOK_APP_SECRET=''
+GOOGLE_APP_ID=''
+GOOGLE_APP_SECRET=''
```
Comment (kcharles52) [inline .env.example]: Add a blank line at the end of the file
```suggestion
GOOGLE_APP_SECRET=''

```

File: src/index.js (hunk 3 above)
Comment (kcharles52) [inline src/index.js]:
```suggestion
export default app;

```

File: src/index.js [hunk 8]
```diff
@@ -1,93 +1,13 @@
...
-const isProduction = process.env.NODE_ENV === "production";
-
-// Create global app object
+dotenv.config();
 const app = express();
+app.get('/', (req, res) => res.status(200).send({
+  message: 'Welcome to Barefoot Nomad.'
+}));
 
-app.use(cors());
```
Comment (kcharles52) [inline src/index.js]: Please restore this  and the following middleware because we need in this project

**Review by kcharles52 — CHANGES_REQUESTED** [review]

**Review by kcharles52 — APPROVED** [review]

**Pull request merged**